gtest/src/system.rs

```diff
@@ -24,15 +24,97 @@ use crate::{
 };
 use colored::Colorize;
 use env_logger::{Builder, Env};
-use gear_core::{ids::CodeId, message::Dispatch};
+use gear_core::{
+    ids::{CodeId, ProgramId},
+    memory::PageBuf,
+    message::Dispatch,
+    pages::GearPage,
+};
+use gear_lazy_pages::{LazyPagesPagesStorage, LazyPagesVersion, PagePrefix, PageSizes};
+use gear_lazy_pages_common::LazyPagesInitContext;
 use path_clean::PathClean;
-use std::{borrow::Cow, cell::RefCell, env, fs, io::Write, path::Path, thread};
+use std::{
+    borrow::Cow,
+    cell::RefCell,
+    collections::{BTreeMap, BTreeSet, HashMap},
+    env, fs,
+    io::Write,
+    path::Path,
+    rc::Rc,
+    thread,
+};
+
+#[derive(Debug, Default, Clone)]
+pub(crate) struct PagesData(Rc<RefCell<HashMap<ProgramId, BTreeMap<GearPage, PageBuf>>>>);
+
+impl PagesData {
+    pub fn insert(&mut self, program_id: ProgramId, page: GearPage, page_data: PageBuf) {
+        self.0
+            .borrow_mut()
+            .entry(program_id)
+            .or_default()
+            .insert(page, page_data);
+    }
+
+    pub fn remove(&mut self, program_id: ProgramId, page: GearPage) {
+        self.0.borrow_mut().entry(program_id).and_modify(|set| {
+            set.remove(&page);
+        });
+    }
+
+    pub fn get(&self, program_id: ProgramId) -> BTreeMap<GearPage, PageBuf> {
+        self.0
+            .borrow()
+            .get(&program_id)
+            .cloned()
+            .unwrap_or_default()
+    }
+
+    pub fn get_page(&self, program_id: ProgramId, page: GearPage) -> Option<PageBuf> {
+        self.0.borrow().get(&program_id)?.get(&page).cloned()
+    }
+
+    pub fn pages_keys(&self, program_id: ProgramId) -> BTreeSet<GearPage> {
+        self.0
+            .borrow()
+            .get(&program_id)
+            .map(|map| map.keys().copied().collect())
+            .unwrap_or_default()
+    }
+}
+
+#[derive(Debug)]
+struct PagesStorage {
+    pages_data: PagesData,
+}
+
+impl LazyPagesPagesStorage for PagesStorage {
+    fn page_exists(&self, prefix: &PagePrefix, page: GearPage) -> bool {
+        let program_id = prefix.program_id();
+        self.pages_data.get_page(program_id, page).is_some()
```

Comment:
intro different method to increase perf?

Reply:
I doubt `gtest` worth it